src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResultSet.java

```diff
@@ -393,8 +393,6 @@ boolean onDone(TDSReader tdsReader) throws SQLServerException {
                     SQLServerException.makeFromDriverError(stmt.connection, stmt, form.format(msgArgs), null, false);
                 }
 
-                stmt.connection.getSessionRecovery().decrementUnprocessedResponseCount();
```

Comment:
This decrement is unnecessary because we are throwing an exception now on doneToken errors. In the old code, when we didn't explicitly throw an exception, we would decrement only on a `DONE_ERROR` eg. tds status flag of 0x0002.

Reply:
this makes me think we need to add that back to the other onDone methods in Statement and CallableStatement as well, and, add tests for them

Reply:
can you add a comment here so we don't think it's missed when we look at this in the future